src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -283,6 +297,21 @@ public ISymbolNode HelperEntrypoint(HelperEntrypoint entrypoint)
             return symbol;
         }
 
+        private TypeDesc _systemICastableType;
+
+        public TypeDesc ICastableInterface
```

Comment:
Well, to structure this better, I would ideally like to see something like the factory have methods only for creating/interning nodes + a single accessor that gets to a dumping ground for stuff like this, but I don't know how valuable that really is. The same dumping ground is present in our previous attempt at this sort of dependency analysis, and while it did grow to about 15 bits of extra data, it wasn't uncontrollable. What I'd say is... eww, but I'm not super concerned.